packages/core/src/utils/index.ts

```diff
@@ -107,15 +109,13 @@ export function defineReadOnly<T, K extends keyof T>(
   });
 }
 
-export function filterUndefined(
-  obj: Record<string, unknown>
-): Record<string, unknown> {
+export function filterUndefined<T>(obj: Deferrable<T>): T {
```

Comment:
The `Deferrable` in the type signature here seems wrong. If some of the values of the object are promises resolving to null, presumably you'd want to remove them right?

If I understand correctly, we don't need it to work with such values since we resolved all the values when we did `await Promise.all` above, so just `obj: T` should be fine.

Reply:
updated